app/services/idv/aamva_state_maintenance_window.rb

```diff
@@ -0,0 +1,165 @@
+        { cron: '0 9 * * Sun', duration: 90 },
+        # ...and 5 am to 5:45 am ET on 2nd Sunday of month.
+        { cron: '0 5 * * Sun#2', duration: 45 },
+      ],
+      'NC' => [
+        # Daily, Midnight to 7:00 am. ET.
+        { cron: '0 0 * * *', duration: 7 * 60 },
+        # Sundays from 5am. till Noon
+        { cron: '0 5 * * Sun', duration: 7 * 60 },
+      ],
+      # NM: "Sunday mornings." (not modeling; too vague)
+      'NY' => [
+        # Sunday maintenance 8 pm. to 9 pm. ET.
+        { cron: '0 20 * * Sun', duration: 60 },
+      ],
+      'PA' => [
+        # Sunday maintenance may occur, often between 5:30 am. & 7:00 am. ET
+        { cron: '30 5 * * Sun', duration: 90 },
+      ],
+      'SC' => [
+        # Sunday maintenance from 7:00 pm. to 10:00 pm. ET.
+        { cron: '0 19 * * Sun', duration: 3 * 60 },
+      ],
+      'TX' => [
+        # Downtime on weekends between 9 pm ET to 7 am ET.
+        { cron: '0 21 * * Sat,Sun', duration: 10 * 60 },
+      ],
+      'VA' => [
+        # Sunday morning maintenance 3:00 am. to 5 am. ET.
+        { cron: '0 3 * * Sun', duration: 120 },
+        # Daily maintenance from 5 am. to 5:30 am.
+        { cron: '0 5 * * *', duration: 30 },
+        # "Might not respond for short spells, daily between 7 pm  and 8:30 pm." (not modeling this)
+      ],
+      'VT' => [
+        # Daily maintenance from midnight to 5 am. ET.
+        { cron: '0 0 * * *', duration: 5 * 60 },
+      ],
+      'WA' => [
+        # Maintenance from Saturday 9:45 pm. to Sunday 8:15 am. ET.
+        { cron: '45 21 * * Sat', duration: 10.5 * 60 },
+      ],
+      'WI' => [
+        # Downtime on Tuesday – Saturday typically between 3 – 4 am ET.
+        { cron: '0 3 * * Tue-Sat', duration: 60 },
+        # Downtime on Sunday from 6 – 10 am. ET.
+        { cron: '0 6 * * Sun', duration: 4 * 60 },
+      ],
+      'WV' => [
+        # Occasional Sunday maintenance from 6:00 am. to noon ET.
+        { cron: '0 6 * * Sun', duration: 6 * 60 },
+      ],
+      'WY' => [
+        # Daily, 2 am. to 5 am. ET.
+        { cron: '0 2 * * *', duration: 3 * 60 },
+      ],
+    }.freeze
+
+    PARSED_MAINTENANCE_WINDOWS = MAINTENANCE_WINDOWS.transform_values do |windows|
+      Time.use_zone(TZ) do
+        windows.collect do |window|
+          cron = Fugit.parse_cron(window[:cron])
+          { cron: cron, duration: window[:duration] }
+        end
+      end
+    end.freeze
+
+    class << self
+      def in_maintenance_window?(state)
+        Time.use_zone(TZ) do
+          windows_for_state(state).any? { |window| window.cover?(Time.zone.now) }
+        end
+      end
+
+      def windows_for_state(state)
+        return [] unless PARSED_MAINTENANCE_WINDOWS[state]
+
+        Time.use_zone(TZ) do
+          PARSED_MAINTENANCE_WINDOWS[state].collect do |window|
+            previous = window[:cron].previous_time.to_t
+            (previous..(previous + window[:duration].minutes))
+          end
+        end
```

Comment:
idea for simpler control flow, fewer exit points

```suggestion
        Time.use_zone(TZ) do
          PARSED_MAINTENANCE_WINDOWS.fetch(state, []).map do |window|
            previous = window[:cron].previous_time.to_t
            (previous..(previous + window[:duration].minutes))
          end
        end
```

also I prefer `#map` over `#collect`, and I think it's much more consistent with the rest of our codebase too

```bash
identity-idp:main> git grep \\.collect -- '*.rb' | wc -l
       2
identity-idp:main> git grep \\.map -- '*.rb' | wc -l
     382
```

Reply:
I like this.

I would not be shocked if the two `.collect` uses were my doing. It seems like it's become an idiosyncrasy of mine. I'm fully on board with going with the `.map` norm, but left to my own devices I guess I find it clearer that .`collect` will, err, collect the results. Your proposed change reads a lot better than what I have, though.

Reply:
I took this as some inspo and now after https://github.com/18F/identity-idp/pull/11144, Rubocop will help us enforce `#map`